packages/core/src/amazonq/messages/messagePublisher.ts

```diff
@@ -12,3 +13,45 @@ export class MessagePublisher<T> {
         this.eventEmitter.fire(event)
     }
 }
+
+/**
+ * Same as {@link MessagePublisher}, but will wait until the UI indicates it
+ * is ready to recieve messages, before the message is published.
+ *
+ * This solves a problem when running a right click menu option like
+ * "Send To Prompt" BUT chat is not opened yet, it would result in the prompt failing to
+ * be recieved by chat.
+ */
+export class UiReadyMessagePublisher<T> extends MessagePublisher<T> {
+    constructor(
+        eventEmitter: EventEmitter<T>,
+        private readonly isChatUiReady: () => boolean
+    ) {
+        super(eventEmitter)
+    }
+    public override publish(event: T): void {
```

Comment:
Does it make sense to encapsulate the logic for isUiReady in this class somehow? Technically, it seems like this class is meant to hold requests until the UI is ready so keeping the uiReady logic close might make more sense

Reply:
Yeah this makes sense, updated